File: summerHelpers/src/main/java/com/summer/helper/dialog/BaseCenterDialog.java
package com.summer.helper.dialog;

import android.content.Context;
import android.support.annotation.NonNull;

import com.malata.summer.helper.R;


/**
 * 中间弹出框基本样式
 * Created by xiastars on 2017/6/20.
 */

public abstract class BaseCenterDialog extends BaseDialog {

    public BaseCenterDialog(@NonNull Context context) {
        super(context, R.style.TagFullScreenDialog);
        this.context = context;
        setCanceledOnTouchOutside(true);
    }

    public BaseCenterDialog(@NonNull Context context, int style) {
        super(context, style);
        this.context = context;
        setCanceledOnTouchOutside(true);
    }

    @Override
    protected int showEnterAnim() {
        return R.anim.dialog_center_enter;
    }

    @Override
    protected int showQuitAnim() {
        return R.anim.dialog_center_quit;
    }

}

File: summerHelpers/src/main/java/com/summer/helper/dialog/BaseCenterDialog.kt
package com.summer.helper.dialog

import android.content.Context
import com.malata.summer.helper.R

/**
 * 中间弹出框基本样式
 * Created by xiastars on 2017/6/20.
 */
abstract class BaseCenterDialog(context: Context) : BaseDialog(context) {

    override fun showEnterAnim(): Int {
        return R.anim.dialog_center_enter
    }

    override fun showQuitAnim(): Int {
        return R.anim.dialog_center_quit
    }
}